### PR TITLE
Support for IUserContextBuilder

### PR DESCRIPTION
**Pull request by fubar-coder**
- Additional reference to the `GraphQL.Server.Transports.AspNetCore` package
- UserContextFactory marked as obsolete
- A configured UserContextFactory takes precedence over a configured IUserContextBuilder

# Important

Merging this PR together with PR #18 results in uncompilable code. The reason for this is, that `GraphQL.Server.Transports.AspNetCore` 5.0.0 only supports `netcoreapp3.1` and `net5.0`.

In other words: The `netstandard2.0`, `netcoreapp2.1` and `netcoreapp3.0` target frameworks have to be removed from the projects `GraphQL.Upload.AspNetCore`, and `GraphQL.Upload.AspNetCore.Tests`.

### CODE REVIEW

**Comment by JannikLassahn** [comment]
I'm not sure if I really want this library to have a dependency on GraphQL.Server.Transports.AspNetCore just to drop the existing user factory. Especially if this means breaking targets that are still supported by GraphQL.

I'll close this for now, but if you have any objections please let me know.

**Comment by fubar-coder** [comment]
What about a custom version of `IUserContextBuilder`. This would allow you to develop some kind of "adapter service", which redirects building the context to the `IUserContextBuilder` from `GraphQL.Server.Transports.AspNetCore`?

**Comment by JannikLassahn** [comment]
I guess having our own interface and getting the respective instance by the service provider is a bit cleaner overall. 
Ultimately, I would like to have a single extension point for providing a user context to minimize confusion and API surface.
That being said, the current method seems to work just fine (even as an "adapter").

**Comment by fubar-coder** [comment]
Sadly, this doesn't work, because the `IUserContextBuilder` has a method `BuildUserContext` with the following signature:

```csharp
Task<IDictionary<string, object>> BuildUserContext(HttpContext httpContext);
```

Using an "adapter function" is not possible, because it'd require configuring `GraphQLUploadOptions` as follows:

```csharp
services.AddOptions<GraphQLUploadOptions>()
    .Configure<IUserContextBuilder>(
        (opt, ucb) =>
        {
            opt.RequestSerializerOptions.ConfigureDefaults();
            opt.UserContextFactory = ctx => /* await */ ucb.BuildUserContext(ctx);
        });
```

Using await isn't possible here, but it'd be required, because `BuildUserContext` returns a task.

**Comment by JannikLassahn** [comment]
I see. In that case I definitely see the value of changing the current implementation.
